src/Controller/FrontendUsersProfileController.php

```diff
@@ -111,23 +115,62 @@ private function getUserRecord(ContentType $contentType): Content
     {
         /** @var User $user */
         $user = $this->getUser();
+        
+        // Access user record, if available
+        $contentTypeSlug = $this->getExtension()->getExtConfig('contenttype', $user->getRoles()[0]);
+        $contentType = $this->getBoltConfig()->getContentType($contentTypeSlug);
+
         $content = $this->contentRepository->findBy([
             'author' => $user,
             'contentType' => $contentType->getSlug(),
         ]);
-
+        
+        // If user record unavailable, create it
         if (empty($content)) {
-            $content = new Content($contentType);
-            $content->setAuthor($user);
-            $content->setPublishedAt(new \DateTime());
-            $content->setStatus(Statuses::PUBLISHED);
-            $contentTypeName = strtolower($contentType->get('name', $contentType->get('slug')));
-            $content->setContentType($contentTypeName);
-            $this->contentFillListener->fillContent($content);
-        } elseif (is_iterable($content)) {
+            return $this->new($contentType);
+        } 
+        elseif (is_iterable($content)) {
             $content = end($content);
         }
+        
+        return $content;
+    }
 
+    // Define the user record content
+    private function new(ContentType $contentType): Content
+    {
+        /** @var User $user */
+        $user = $this->getUser();
+        $contentTypeSlug = $this->getExtension()->getExtConfig('contenttype', $user->getRoles()[0]);
+        $contentType = $this->getBoltConfig()->getContentType($contentTypeSlug);
+        
+        $content = new Content($contentType);
+        $content->setAuthor($user);
+        $content->setCreatedAt(new \DateTime());
+        $content->setPublishedAt(new \DateTime());
+        $content->setStatus(Statuses::PUBLISHED);
```

Comment:
Maybe this was already an issue before (so could be out of scope?) but we have this `initial_status` option in the `config.yaml`. Is this the right place to make sure of that, instead of hardcoding it to publsihed?

Reply:
As for this bit, @I-Valchev, it appears that we really do not need to hardcode the `published` status. Even completely removing `$content->setStatus(Statuses::PUBLISHED);` on line 15 does not affect the publising status. The record is still published.

Reply:
Nice, thanks!

Reply:
Sure thing!